libraries/common/src/main/java/androidx/media3/common/MimeTypes.java

```diff
@@ -590,6 +590,10 @@ public static String getMimeTypeFromMp4ObjectType(int objectType) {
         return C.ENCODING_DTS;
       case MimeTypes.AUDIO_DTS_HD:
         return C.ENCODING_DTS_HD;
+      case MimeTypes.AUDIO_DTS_EXPRESS:
+        return C.ENCODING_DTS; // This should be ENCODING_DTS_HD when IC platforms support it.
```

Comment:
Does comment mean if the platform doesn't support `ENCODING_DTS_HD`, then it can be treated as `ENCODING_DTS`? 

Maybe we can directly return `C.ENCODING_DTS_HD` here, because when doing the actual format support querying, we will look at if `C.ENCODING_DTS_HD` is supported according to the audio capabilities. If not, we will treat it as the `C.ENCODING_DTS` - https://github.com/androidx/media/blob/1ef427f640277dc8e0ad09ebf86be972f216eb82/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/AudioCapabilities.java#L198-L200

Reply:
Currently, to playback DTS audio in tunnel mode via Direct playback mode, it is better to create an AudioTrack with ENCODING_DTS regardless of whether the incoming DTS stream is  DTS Digital surround, DTS Express or DTS:X Profile 2. I know creating AudioTrack on some MTK platforms with ENCODING_DTS_HD will cause a crash. We are working with IC vendors to resolve this. So in future we should be using ENCODING_DTS_HD for DTS Express. But for now, it should be only ENCODING_DTS. Thanks.

Reply:
I see. And I saw the offload logic is also [calling this method].(https://github.com/androidx/media/blob/release/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/DefaultAudioSink.java#L1716), so any change may potentially impact the offload case.

If we return `ENCODING_DTS_HD` and it is supported by the platform for offload playback, then an `AudioTrack` will be [tried to create with `ENCODING_DTS_HD`](https://github.com/androidx/media/blob/release/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/audio/DefaultAudioSink.java#L762-L763), and leads to the MTK platform issue you mentioned. The current implementation may be the simplest way to care about both passthrough and offload cases.

cc: @christosts

Reply:
@tianyif Anyway, DTS_Express is supposed to be transcoded to DTS Digital Surround. We do not allow passthrough of DTS Express to audio sink devices directly. In my experience, the support for DTS Express using ENCODING_DTS_HD AudioTrack is inconsistent. To solve all these issues, we use tunnel mode with direct playback for DTS Express. The downstream platform's DSP will take care of decoding/passthrough/transcoding. So an application such as Exoplayer just needs to send the frames downstream to the platform's DSP (Decoder) via AudioTrack.